### ISSUE TITLE
Bug: [space-infix-ops] false positive with optional property since version 5.27.0

### ISSUE THREAD

**Issue by breilly-athenium** [typescript-eslint/typescript-eslint#5154]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=MYGwhgzhAECiAeYC2AHEBTaBvAUNaA9igC4CWBAdmCAPzQC80AZtROgNx7QD03+-0AHrQA8inQAnMMQIToAchrzoSAK4Ri0AEaYIKMMHQATAHQ4AvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKWQBwEMBjRAWgEsA7AM3IA9SB7PVDSR66yAGnG0gACAFwCeeFEWjk8Q0inhUhAenzEyVWg2asoiaNEbRIfAL4gTQA&tsconfig=N4XyA

### Repro Code

```typescript
class Example {
  optional? = false;
  //      ^ Operator '?' must be spaced.
}
```

### ESLint Config

```javascript
{
  "rules": {
    "space-infix-ops": "off",
    "@typescript-eslint/space-infix-ops": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

no error

```ts
class Example {
  optional? = false;
  //      ^ Operator '?' must be spaced.

  // no error when explicitly typed or untyped
  explicitlyTypedOptionalWithInitialValue?: boolean = false;
  explicitlyTypedOptional?: boolean;
  untypedOptional?;
}
```

### Actual Result

`2:11 error Operator '?' must be spaced. @typescript-eslint/space-infix-ops`

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.27.1` |
| `@typescript-eslint/parser`            | `5.27.1` |
| `TypeScript`                           | `4.7.2` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `14.19.3` |
